### PR TITLE
Add tests for JMX Quartz Scheduler management

### PR DESCRIPTION
**Pull request by gmunozfe**
Added a springboot sample to manage Quartz Scheduler through JMX.

These scenarios may be useful for handling blue-green deployments, when one of the instances needs to be paused or even restarted.

### CODE REVIEW

File: kie-spring-boot/kie-spring-boot-samples/jbpm-spring-boot-sample-basic/src/test/java/org/jbpm/springboot/samples/JMXQuartzSchedulerTest.java [hunk 1]
```diff
@@ -0,0 +1,205 @@
+package org.jbpm.springboot.samples;
```
Comment (sutaakar) [inline kie-spring-boot/kie-spring-boot-samples/jbpm-spring-boot-sample-basic/src/test/java/org/jbpm/springboot/samples/JMXQuartzSchedulerTest.java]: Please add licence header.

Comment (gmunozfe) [reply kie-spring-boot/kie-spring-boot-samples/jbpm-spring-boot-sample-basic/src/test/java/org/jbpm/springboot/samples/JMXQuartzSchedulerTest.java]: Thanks! Added

**Comment by gmunozfe** [comment]
jenkins retest this

File: kie-spring-boot/kie-spring-boot-samples/jbpm-spring-boot-sample-basic/src/test/java/org/jbpm/springboot/samples/JMXQuartzSchedulerTest.java [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+		countDownListener.configure(PROCESS_ID, 1);
+
+		long processInstanceId = processService.startProcess(unit.getIdentifier(), PROCESS_ID);
+		assertNotNull(processInstanceId);
+
+		pauseScheduler();
+		
+		// Wait 20 seconds, as the timer was scheduled for 15 seconds
+		CountDownLatch latch = new CountDownLatch(1);
+		latch.await(20, TimeUnit.SECONDS);
```
Comment (MarianMacik) [inline kie-spring-boot/kie-spring-boot-samples/jbpm-spring-boot-sample-basic/src/test/java/org/jbpm/springboot/samples/JMXQuartzSchedulerTest.java]: Same here. And later you can again call await on the same countDownListener.

File: kie-spring-boot/kie-spring-boot-samples/jbpm-spring-boot-sample-basic/src/test/java/org/jbpm/springboot/samples/JMXQuartzSchedulerTest.java [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+	@Test(timeout = 30000)
+	public void whenSchedulerPausedThenTimerNotFired() throws Exception {
+		long processInstanceId = processService.startProcess(unit.getIdentifier(), PROCESS_ID);
+		assertNotNull(processInstanceId);
+
+		pauseScheduler();
+
+		// Wait 20 seconds, as the timer was scheduled for 15 seconds
+		CountDownLatch latch = new CountDownLatch(1);
+		latch.await(20, TimeUnit.SECONDS);
```
[the remark this answers is not in the document]
Comment (gmunozfe) [reply kie-spring-boot/kie-spring-boot-samples/jbpm-spring-boot-sample-basic/src/test/java/org/jbpm/springboot/samples/JMXQuartzSchedulerTest.java]: Thanks Marián, good suggestion: now reusing countdownlatch listener in a split method, there's a single point for awaiting without interruption (process has not finished yet) at both tests.

File: kie-spring-boot/kie-spring-boot-samples/jbpm-spring-boot-sample-basic/src/test/java/org/jbpm/springboot/samples/JMXQuartzSchedulerTest.java (hunk 2 above)
Comment (gmunozfe) [reply kie-spring-boot/kie-spring-boot-samples/jbpm-spring-boot-sample-basic/src/test/java/org/jbpm/springboot/samples/JMXQuartzSchedulerTest.java]: OK, same resolution, thanks!

**Comment by MarianMacik** [comment]
@sutaakar Can we proceed with merge?

**Comment by sutaakar** [comment]
Sure, looks good for me.

**Comment by MarianMacik** [comment]
@gmunozfe I just changed indentation to spaces. Didn't want to bother you with that change before the weekend. Will merge as soon as the test run is green. In the meantime, please make sure that you configure a space as a default indentation character and not a tab. It then causes unwanted changes in diffs if somebody uses this and somebody else that.

**Comment by gmunozfe** [comment]
Thanks @MarianMacik, I wasn't aware that the formatter profile was per workspace in Eclipse. Fixed.

**Comment by manstis** [comment]
There's code formatters for Eclipse and InteliJ in `droolsjbpm-build-bootstrap`.

Everyone is encouraged to use them to prevent formatting differences poluting diffs.

**Comment by MarianMacik** [comment]
jenkins retest this

**Comment by MarianMacik** [comment]
jenkins retest this

**Pull request merged**